### PR TITLE
Pawn move

### PR DESCRIPTION
**Pull request by ghost**
Hey Diane, did you have a branch where you put in the moves for the rook and knight? I thought you did, but can't find it. 

The pawn is turning out to be harder than I thought. How do I make it so that the white pawn can only move +1 on the y axis and the black pawn can only move -1 on the y axis?

### CODE REVIEW

**Comment by ghost** [comment]
Oh my god, I'm an idiot Diane. We worked on it and merged into the master. I see all your logic for all the other pieces. I'm going to refactor the pawn to take advantage of your legal_vert_move method in your piece.rb model

**Comment by washburnad** [comment]
@davidslee101 this definitely needs some tests.  Include tests you expect to fail (moving backward, diagonal, sideways, etc...)

**Comment by ghost** [comment]
arghh i keep forgetting tests. 

i was more concerned with trying to figure out how to make it so that the white pawn can only move forwards and the black pawn can only move backwards.

I think I need to use if/else statements.

**Comment by washburnad** [comment]
write the tests - watch them fail - then write the code so that they pass.

**Comment by ghost** [comment]
Diane and I put in awesome tests (thanks Aaron!)
Pawn is ready to go.

**Comment by dvanetten03** [comment]
Went through all tests and pawn.rb.  Looks awesome, great code, David!

File: app/models/pawn.rb [hunk 1]
```diff
@@ -4,9 +4,24 @@ def puts_piece_type
...
   end
 
-  def valid_move?
-    # put code for how this piece should move here
+  def legal_move?(row,col)
+    current_row = self.row_position
+    current_col = self.col_position
 
+    row_dif = (current_row - row)
+    col_dif = (current_col - col)
```
Comment (washburnad) [inline app/models/pawn.rb:12]: Since we only use current_row/col once we can merge lines 8/9 into lines 11/12

File: app/models/pawn.rb [hunk 2]
```diff
@@ -4,9 +4,24 @@ def puts_piece_type
...
+
+
+    #this checks to see if white pawn is moving forward on the board's column
+   if row >= 0 and row <= 7 and col >= 0 and col <= 7
```
Comment (washburnad) [inline app/models/pawn.rb:16]: As I recall, we were pulling this out into a separate piece.rb method.  on_board? or something like that.

File: test/models/pawn_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    pawn = FactoryGirl.create(:pawn, :color => "White", :row_position => 1, :col_position => 0)
+    assert pawn.legal_move?(2, 0)
+    # test backward movement
+    refute pawn.legal_move?(0, 0)
+    # test horizontal movement
+    refute pawn.legal_move?(1, 1)
+    # test diagonal movement
+    refute pawn.legal_move?(2, 1)
+    # test two square movement
+    refute pawn.legal_move?(3, 0)
```
Comment (washburnad) [inline test/models/pawn_test.rb:14]: We will eventually want to assert the test on line 14.  Might be good to add a TODO in the comment on line 13.
We might want to add a 2 square refute test with the pawn in a different location.

**Pull request merged**